src/generate.js

```diff
@@ -16,14 +16,50 @@
 
 //@ts-check
 
+const fs = require('fs');
 const path = require('path');
+const https = require('https');
 const { Generator } = require('./generator');
 const { JavaScriptFormatter } = require('./format_js');
 const { PythonFormatter } = require('./format_python');
 const { JavaFormatter } = require('./format_java');
 const { CSharpFormatter } = require('./format_csharp');
 
-new Generator('js', path.join(__dirname, '..', 'nodejs', 'docs'), new JavaScriptFormatter());
-new Generator('python', path.join(__dirname, '..', 'python', 'docs'), new PythonFormatter());
-new Generator('java', path.join(__dirname, '..', 'java', 'docs'), new JavaFormatter());
-new Generator('csharp', path.join(__dirname, '..', 'dotnet', 'docs'), new CSharpFormatter());
+(async () => {
+  new Generator('js', path.join(__dirname, '..', 'nodejs', 'docs'), new JavaScriptFormatter());
+  new Generator('python', path.join(__dirname, '..', 'python', 'docs'), new PythonFormatter());
+  new Generator('java', path.join(__dirname, '..', 'java', 'docs'), new JavaFormatter());
+  new Generator('csharp', path.join(__dirname, '..', 'dotnet', 'docs'), new CSharpFormatter());
+
+  await updateStarsButton();
+})().catch(error => {
+  console.error(error);
+  process.exit(error);
+});
+
+async function updateStarsButton() {
+  const kGitHubStarsButtonSource = path.join(__dirname, 'components/GitHubStarButton/index.tsx');
```

Comment:
let's add a comment around the line in `components/GitHubStarButton/index.tsx` that the number is generated by this script, so noone should change manually or edit that line.

Like:

```
STARS = '37k'; // NOTE: this line is generated by src/generate.js. Do not change!
```

Then later you'd be able to edit it easier without the regex:

```js
const lines = content.split('\n');
const MAGIC_COMMENT = '// NOTE: this line is generated by src/generate.js. Do not change!';
const starLineIndex = lines.findIndex(line => line.includes(MAGIC_COMMENT));
lines[starLineIndex] = `STARS='${count}k'; ${MAGIC_COMMENT}`;
```